wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/provision/env/CheApiEnvVarProvider.java

```diff
@@ -18,24 +18,21 @@
 
 /**
  * CHE_API endpoint var provided. For all currently supported infrastructures, it reuses {@link
- * CheApiInternalEnvVarProvider} to provide the value.
+ * CheApiExternalEnvVarProvider} to provide the value.
  *
- * @deprecated this class shall soon be removed, as this variable is provided only for backward
- *     compatibility
  * @author Sergii Leshchenko
  * @author Mykhailo Kuznietsov
  */
-@Deprecated
```

Comment:
I'm not convinced this class shouldn't still be deprecated, since having three providers with some interdependency is strange. Which of `CHE_API|CHE_API_INTERNAL|CHE_API_EXTERNAL` are currently used? IMO only one of `CHE_API` or `CHE_API_EXTERNAL` should be provisioned.

Ideally I would like to see `CHE_API` (the external, publicly accessible URL) and `CHE_API_INTERNAL` used, with `CHE_API_EXTERNAL` removed.

Reply:
Yes `CHE_API_EXTERNAL` is redundant. The reason I left it is that Che-Theia uses it now.

https://github.com/eclipse/che-theia/blob/546916ad2fdd058a37f31f7bea6d324591c7ae20/extensions/eclipse-che-theia-terminal/src/node/workspace-service-impl.ts#L106

I have no objection to set the deprecation flag to `CHE_API_EXTERNAL` if we agreed.

Reply:
@monaka I suppose in that case it's safer to leave all three for now and plan deprecation in a separate PR/issue.